CSS Keywords should be lowercased during serialization

cdumez opened the pull request:
#### 5c14835fdcfe302f7e5b85d732ab092676799325
<pre>
CSS Keywords should be lowercased during serialization
<a href="https://bugs.webkit.org/show_bug.cgi?id=249438">https://bugs.webkit.org/show_bug.cgi?id=249438</a>

Reviewed by Sam Weinig.

CSS Keywords should be lowercased during serialization:
- <a href="https://www.w3.org/TR/cssom-1/#serialize-a-css-component-value">https://www.w3.org/TR/cssom-1/#serialize-a-css-component-value</a>

* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/color-interpolation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/pointer-events-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/shape-rendering-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/text-rendering-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-presentation-attribute-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/parsing/color-interpolation-filters-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/parsing/color-interpolation-filters-parsing-valid-expected.txt:
Rebaseline WPT tests now that more checks are passing.

* Source/WebCore/css/CSSPrimitiveValue.cpp:
(WebCore::CSSPrimitiveValue::formatNumberForCustomCSSText const):
Call nameStringForSerialization() instead of nameString() since we are serializing.
This makes sure that the identifier is serialized as lowercase.

* Source/WebCore/css/typedom/CSSStyleValueFactory.cpp:
(WebCore::CSSStyleValueFactory::reifyValue):
Per <a href="https://drafts.css-houdini.org/css-typed-om-1/#reify-ident">https://drafts.css-houdini.org/css-typed-om-1/#reify-ident</a>, we should create
a CSSKeywordValue from the *serialization* of the identifier, which means that
the identifier would become lowercase.

* Source/WebCore/css/process-css-values.py:
(GenerationContext):
Generate nameStringForSerialization() which provides the name string of the
keyword as lowercase, for the purpose of serialization.

Canonical link: <a href="https://commits.webkit.org/258024@main">https://commits.webkit.org/258024@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0cc1187a5b78dde8df1974b3c91a878798d1d52a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100648 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9791 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33687 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109949 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/170225 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/104631 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10726 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/316 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93046 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107797 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106429 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/91354 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34715 "Passed tests") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22746 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/77684 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3496 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24274 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3512 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/316 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9624 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43771 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5503 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5302 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WebCore/css/process-css-values.py:
```diff
@@ -243,6 +258,18 @@ def _generate_lookup_functions(self, *, to):
                 return string;
             }
 
+            const AtomString& nameStringForSerialization(CSSValueID id)
```
Mind adding a link to the spec text that says to lowercase it somewhere - https://drafts.csswg.org/cssom/#serialize-a-css-component-value

cdumez replied:
It's in the commit log but you're right, I should have it as a code comment as well. Will add.

cdumez committed:
CSS Keywords should be lowercased during serialization
https://bugs.webkit.org/show_bug.cgi?id=249438

Reviewed by Sam Weinig.

CSS Keywords should be lowercased during serialization:
- https://www.w3.org/TR/cssom-1/#serialize-a-css-component-value

* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/color-interpolation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/pointer-events-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/shape-rendering-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/text-rendering-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-presentation-attribute-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/parsing/color-interpolation-filters-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/parsing/color-interpolation-filters-parsing-valid-expected.txt:
Rebaseline WPT tests now that more checks are passing.

* Source/WebCore/css/CSSPrimitiveValue.cpp:
(WebCore::CSSPrimitiveValue::formatNumberForCustomCSSText const):
Call nameStringForSerialization() instead of nameString() since we are serializing.
This makes sure that the identifier is serialized as lowercase.

* Source/WebCore/css/typedom/CSSStyleValueFactory.cpp:
(WebCore::CSSStyleValueFactory::reifyValue):
Per https://drafts.css-houdini.org/css-typed-om-1/#reify-ident, we should create
a CSSKeywordValue from the *serialization* of the identifier, which means that
the identifier would become lowercase.

* Source/WebCore/css/process-css-values.py:
(GenerationContext):
Generate nameStringForSerialization() which provides the name string of the
keyword as lowercase, for the purpose of serialization.

Canonical link: https://commits.webkit.org/258024@main
